apps/03TicTacToe.js

```diff
@@ -23,24 +23,41 @@ function printBoard() {
 
 function horizontalWin() {
     // Your code here
+    return (board[0][0] === playerTurn && board[0][1] === playerTurn && board[0][2] === playerTurn) ||
+    (board[1][0] === playerTurn && board[1][1] === playerTurn && board[1][2] === playerTurn) ||
+    (board[2][0] === playerTurn && board[2][1] === playerTurn && board[2][2] === playerTurn) ;
+
 }
 
 function verticalWin() {
     // Your code here
+    return (board[0][0] === playerTurn && board[1][0] === playerTurn&& board[2][0] === playerTurn) ||
+    (board[0][1] === playerTurn && board[1][1] === playerTurn&& board[2][1] === playerTurn) ||
+    (board[0][2] === playerTurn && board[1][2] === playerTurn&& board[2][2] === playerTurn);
+
 }
 
 function diagonalWin() {
     // Your code here
+  return (board[0][0] === playerTurn && board[1][1] === playerTurn && board[2][2] ===  playerTurn) ||
+  (board[0][2] === playerTurn && board[1][1] === playerTurn  && board[0][0] === playerTurn);
+
 }
 
 function checkForWin() {
-    // Your code here
+    return (horizontalWin() || verticalWin() || diagonalWin())
+  // Your code here
 }
 
 function ticTacToe(row, column) {
-    // Your code here
+  board[row][column] = playerTurn;
+  checkForWin();
```

Comment:
Everything looks great! I'm going to give you full credit in the grade book.

One thing, there is a requirement to `console.log()` a message when a player wins.

You could do something like this:

```javascript
if (checkForWin()) {
  console.log('Player ' + playerTurn + ' Won!')
}
```